buildhat/devices.py

```diff
@@ -274,3 +275,28 @@ def callback(self, func):
             self._conn.callit = None
         else:
             self._conn.callit = weakref.WeakMethod(func)
+
+    @property
+    def interval(self):
+        """Interval between data points in milliseconds
+
+        :getter: Gets interval
+        :setter: Sets interval
+        :return: Device interval
+        :rtype: int
+        """
+        return self._interval
+
+    @interval.setter
+    def interval(self, value):
+        """Interval between data points in milliseconds
+
+        :param value: Interval
+        :type value: int
+        :raises DeviceError: Occurs if invalid interval passed
+        """
+        if isinstance(value, int):
```

Comment:
Just to check, is there a min and max number we need to check for here?

Reply:
Good point! Will make it check it's >= 0

Reply:
Just changed to 
```
        if isinstance(value, int) and value >= 0 and value <= 1000000000:
```
As they're the values that the firmware will clamp to internally